main.go

```diff
@@ -96,25 +117,46 @@ func (m *MozProcessStat) Add(data MozProcessStat) {
 	m.NetworkIO.PacketsRecv += data.NetworkIO.PacketsRecv
 }
 
-// Diff the provided MozProcessStat to the current one.
-func (m *MozProcessStat) Diff(data MozProcessStat) {
+// StatDiff the provided MozProcessStat to the current one.
+func flattenStat(prev, current MozCollectedStat) FlatMozProcessStat {
 	// Memory fields are absolute, not a sum, so don't diff those.
+	newStat := FlatMozProcessStat{}
+
+	newStat.Timestamp = current.Timestamp
+	newStat.MemoryUsedPercent = current.MemoryUsedPercent
+	newStat.ProcessCount = current.ProcessCount
+	newStat.ThreadCount = current.ThreadCount
+	newStat.AvailableMemory = current.AvailableMemory
+
+	for pid, currentProcess := range current.Processes {
+		// 0-defaults mean we don't worry if we've not seen it before
+		prevProcess, ok := prev.Processes[pid]
+		if ok != true {
+			prevProcess = &MozProcessStat{}
+		}
 
-	m.CPU.User -= data.CPU.User
-	m.CPU.System -= data.CPU.System
-	m.CPU.Idle -= data.CPU.Idle
-	m.CPU.Iowait -= data.CPU.Iowait
-	m.CPU.Steal -= data.CPU.Steal
-
-	m.DiskIO.ReadCount -= data.DiskIO.ReadCount
-	m.DiskIO.WriteCount -= data.DiskIO.WriteCount
-	m.DiskIO.ReadBytes -= data.DiskIO.ReadBytes
-	m.DiskIO.WriteBytes -= data.DiskIO.WriteBytes
-
-	m.NetworkIO.BytesSent -= data.NetworkIO.BytesSent
-	m.NetworkIO.BytesRecv -= data.NetworkIO.BytesRecv
-	m.NetworkIO.PacketsSent -= data.NetworkIO.PacketsSent
-	m.NetworkIO.PacketsRecv -= data.NetworkIO.PacketsRecv
+		newStat.Memory.RSS += currentProcess.Memory.RSS
+		newStat.Memory.VMS += currentProcess.Memory.VMS
+		newStat.Memory.Swap += currentProcess.Memory.Swap
+
+		newStat.CPU.User += currentProcess.CPU.User - prevProcess.CPU.User
```

Comment:
I really like the naming of the `current` and `prev` here, makes it easier to understand it's diff-ing counters to get the actual value 👍